src/uri.cr

```diff
@@ -428,52 +427,51 @@ class URI
     end
   end
 
-  # The global registry for URI scheme default ports.
-  @@default_port = DefaultPort.new
-
-  # Returns the global registry for URI schemes and their respective
-  # default ports.
-  #
-  # The registry can then be used to query the default `port` for a
-  # given `scheme`:
-  #
-  # ```
-  # URI.default_port["http"]  # => 80
-  # URI.default_port["ponzi"] # => nil
-  # ```
-  #
-  # Or it can be used to register the default `port` for a given
-  # `scheme`:
-  #
-  # ```
-  # URI.default_port["ponzi"] = 9999
-  # URI.default_port["ponzi"] # => 9999
-  # ```
-  #
-  # Or it can be used to unregister the default `port` for a given
-  # `scheme`:
+  # A map of schemes and their respective default ports, seeded
+  # with some well-known schemes.
+  @@default_ports = {
+    "ftp"    => 21,
+    "ftps"   => 990,
+    "gopher" => 70,
+    "http"   => 80,
+    "https"  => 443,
+    "ldap"   => 389,
+    "ldaps"  => 636,
+    "nntp"   => 119,
+    "scp"    => 22,
+    "sftp"   => 22,
+    "ssh"    => 22,
+    "telnet" => 23,
+  }
+
+  # Returns the default port for the given scheme if known, otherwise
+  # returns nil.
   #
   # ```
-  # URI.default_port.delete "ftp" # => 21
-  # URI.default_port["ftp"]       # => nil
+  # URI.default_port "http" # => 80
   # ```
-  def self.default_port
-    @@default_port
+  def self.default_port(scheme : String) : Int32?
+    @@default_ports[scheme.downcase]?
   end
 
-  # Returns the default `port` for the given `scheme` if known, otherwise
-  # returns `nil`.
+  # Registers the default port for the given scheme.
+  #
+  # If port is nil, the existing default port for the scheme, if any,
```

Comment:
I think the doc formatting should be ``If *port* is `nil`, ...``

Reply:
Done!